### PR TITLE
Prevent email UIA failures from raising a LoginError

### PR DESCRIPTION
**Pull request by anoadragon453**
Context, Fixes: https://github.com/matrix-org/synapse/issues/9263

In the past to fix an issue with old Riots re-requesting threepid validation tokens, we raised a `LoginError` during UIA instead of `InteractiveAuthIncompleteError`. This is now breaking the way Tchap logs in - which isn't standard, but also isn't disallowed by the spec.

An easy fix is just to remove the 4 year old workaround.

### CODE REVIEW

File: changelog.d/9265.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Prevent password hashes from getting dropped if a client failed threepid validation during a User Interactive Auth stage.
```
Comment (clokep) [inline changelog.d/9265.bugfix:1]: Does it make sense to mention that this removes a workaround for Element < *some number*?

Comment (clokep) [reply changelog.d/9265.bugfix:1]: Or at least "for old Riots"?

**Review by clokep — APPROVED** [review]
LGTM minus the changelog entry.

File: changelog.d/9265.bugfix (hunk 1 above)
Comment (anoadragon453) [reply changelog.d/9265.bugfix:1]: Oh, indeed yes. The fix was added in Riot v0.7.4. I'll add that to the changelog.

**Pull request merged**